internal/backends/winit/glwindow.rs

```diff
@@ -380,34 +415,11 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WindowAdapterSealed for GLWind
     }
 
     fn show(&self) {
-        // Creating a new winit::Window requires access to the running event loop (`&EventLoopWindowTarget`), which
-        // is only possible from within the event loop handler with wasm, as `run()` consumes the event loop. Work
-        // around by invoking `show()` from within the event loop handler.
-        #[cfg(target_arch = "wasm32")]
-        corelib::api::invoke_from_event_loop({
-            let self_weak = send_wrapper::SendWrapper::new(self.self_weak.clone());
-
-            move || {
-                if let Some(this) = self_weak.take().upgrade() {
-                    this.show_impl();
-                }
-            }
-        });
-        #[cfg(not(target_arch = "wasm32"))]
-        self.show_impl();
+        self.call_with_event_loop(Self::show_impl);
     }
 
     fn hide(&self) {
-        self.unmap();
-
-        /* FIXME:
-        if let Some(existing_blinker) = self.cursor_blinker.borrow().upgrade() {
-            existing_blinker.stop();
-        }*/
-        crate::event_loop::with_window_target(|event_loop| {
-            event_loop.event_loop_proxy().send_event(crate::event_loop::CustomEvent::WindowHidden)
-        })
-        .unwrap();
+        self.call_with_event_loop(Self::hide_impl);
```

Comment:
Personal style: I would have use a capture here instead of another function.  Because then the hiding code is in the hide function and one does not have to navigate all around in the file to find the implementation.  But feel free to ignore this.

Reply:
Good that you mention that. I'm a bit torn about this and also don't quite like this. I'll add a patch on top to see how it looks like with a closure.

Reply:
It can stay a `fn(&Self)`  but you can pass a  `|self_| { ... }`  lambda function